src/util/file-filter.js

```diff
@@ -7,24 +7,22 @@ import {createLogger} from './logger';
 
 const log = createLogger(__filename);
 
-// Use this function to mimic path.resolve without resolving to absolute path.
-export const normalizeResolve = (file: string): string => {
-  // normalize
-  file = path.normalize(file);
-
-  // trim trailing slash
-  if (path.parse(file).base && file.endsWith(path.sep)) {
-    return file.slice(0, -1);
+// check if target is a sub directory of src
+export const isSubDir = (src: string, target: string): boolean => {
+  const relate = path.relative(src, target);
+  // same dir
+  if (!relate) {
+    return false;
```

Comment:
@kumar303 @eight04 we should probably raise a UsageError if the artifact dir and the source dir are the same, but I'm not sure that it is nice to raise it from this utility funtion. (it would be probably nicer to raise the UsageError in the FileFilter constructor to make it more explicit).